packages/data/src/world/mq/shadow_temple_mq.yml

```
@@ -114,46 +254,201 @@
+    "Shadow Temple Area Above Shortcut Enclosure": "hookshot_anywhere || climb_anywhere"
+    "Shadow Temple On The Boat": "hookshot_anywhere || climb_anywhere" #Climb Anywhere works on the wheel. TODO: Add as trick - "Adult can jump onto the wheel and climb from there."
+  events:
+    SHADOW_MQ_SHORTCUT_BLOCK_PUSHED: "has_goron_bracelet"
+"Shadow Temple Area Above Shortcut Enclosure":
+  dungeon: Shadow
+  exits:
+    "Shadow Temple Area Before Boat Ride": "true"
+    "Shadow Temple Boarding Boat Landing Area": "can_longshot"
+    "Shadow Temple On The Boat": "hookshot_anywhere"
+  locations:
+    "MQ Shadow Temple Heart Shortcut 1": "true"
+    "MQ Shadow Temple Heart Shortcut 2": "true"
+"Shadow Temple Boarding Boat Landing Area":
+  dungeon: Shadow
+  exits:
+    "Shadow Temple Area Before Boat Ride": "true"
+    "Shadow Temple On The Boat": "true"
+    "Shadow Temple Area Above Shortcut Enclosure": "scarecrow_longshot || longshot_anywhere"
+    #"Shadow Temple Area After Boat Initial Landing": "false" #Maybe with bomb hovering glitches later.
+"Shadow Temple On The Boat": #Made its own region for the Stalfos.
+  dungeon: Shadow
+  exits:
+    "Shadow Temple Area Before Boat Ride": "true"
+    "Shadow Temple Boarding Boat Landing Area": "true"
+    "Shadow Temple Area Above Shortcut Enclosure": "hookshot_anywhere || scarecrow_longshot"
+    "Shadow Temple Area After Boat Initial Landing": "event(SHADOW_MQ_ACTIVATE_BOAT_RIDE)"
+  events:
+    SHADOW_MQ_ACTIVATE_BOAT_RIDE: "can_play_zelda"
+"Shadow Temple Area After Boat Initial Landing":
   dungeon: Shadow
   exits:
-    "Shadow Temple Boss": "boss_key(BOSS_KEY_SHADOW) && (can_use_bow || longshot_anywhere || climb_anywhere)"
-    "Shadow Temple Final Side Rooms": "(can_use_bow && ((can_play_time && can_longshot) || hookshot_anywhere)) || longshot_anywhere || climb_anywhere"
+    #"Shadow Temple On The Boat": "false"
+    #"Shadow Temple Boarding Boat Landing Area": "false" #Maybe with bomb hovering glitches later.
+    "Shadow Temple Invisible Wall Maze Corridor To After Boat": "event(SHADOW_MQ_AREA_AFTER_BOAT_AFTER_BRIDGE_HIGH_LEDGE_SWITCH)"
+    "Shadow Temple Area After Boat After Bridge": "event(SHADOW_MQ_AFTER_BOAT_BRIDGE_FALL) || (can_longshot && event(SHADOW_MQ_AFTER_BOAT_AFTER_BRIDGE_EYE_SWITCH)) || longshot_anywhere || climb_anywhere"
+    "Shadow Temple Area After Boat After Bridge High Ledge": "longshot_anywhere || climb_anywhere"
+  events:
+    SHADOW_MQ_AFTER_BOAT_BRIDGE_FALL: "can_use_bow"
   locations:
-    "MQ Shadow Temple GS After Boat": "gs"
-    "MQ Shadow Temple GS Pre-Boss": "gs && can_use_bow"
+    "MQ Shadow Temple GS After Boat": "gs && (can_hookshot || (can_boomerang && event(SHADOW_MQ_AFTER_BOAT_BRIDGE_FALL)))" #The angle is a little tricky though, but totally doable. Boomerang, though, is possible without the bridge but should be a trick. TODO: 2 tricks: boomerang no bridge, and falling to skultulla after killing it.
```

This should be correct.
```suggestion
    "MQ Shadow Temple GS After Boat": "gs && (can_hookshot || can_use_bow || can_use_slingshot || can_use_din || has_explosives || (can_boomerang && event(SHADOW_MQ_AFTER_BOAT_BRIDGE_FALL)))" #The angle is a little tricky though, but totally doable. Boomerang, though, is possible without the bridge but should be a trick. TODO: trick: boomerang no bridge
```

---

Changed, also added a note about the falling.